### PR TITLE
adds citation form

### PR DESCRIPTION
**Pull request by TheDancingClown**
https://drive.google.com/drive/folders/1uhYfaoLFho69Uzvi9oE0YII3MxIsy-TQ
https://app.asana.com/0/1200175839265057/1200311196271650

- Adds citation model and controller in the group leader namespace
- Adds citation edit form
- Adds helper method for citation max character count dependent on group name length.

Screenshot 2021-08-09 at 00 39 57

- Timestamps submitted date (to display status tag on dashboard)
- Adds placeholder content for dashboard (needs links to publicity guidance, deadlines updating, and link to palace invite)

Screenshot 2021-08-09 at 00 41 26

### CODE REVIEW

File: app/views/group_leader/dashboard/show.html.slim [hunk 1]
```diff
@@ -1,7 +1,95 @@
...
+      | &mdash; awards officially announced
+    p.govuk-body
+      | Successful groups, including your group, are announced in Gazette, GOV.UK and on the QAVS website.
+    h3.govuk-heading-s
+      => deadline_for("buckingham_palace_attendees_details", "%B to")
+      => deadline_for("buckingham_palace_media_information", "%B %Y")
+      | &mdash; Lord-Lieutenant presents the Award
+    p.govuk-body
+      | The Lord-Lieutenant of your local county, as The Queen's representative, will present your group with a certificate and a commemorative crystal at a local ceremony. They will be in touch to arrange a suitable date and time for you.
+    h3.govuk-heading-s
```
Comment (mauricioac) [inline app/views/group_leader/dashboard/show.html.slim:91]: 'govuk-heading-s' maps to h4, not h3

File: app/views/group_leader/dashboard/show.html.slim [hunk 2]
```diff
@@ -1,7 +1,95 @@
...
+          | Incomplete
+    p.govuk-body
+      | Two representatives from the awarded groups can attend a Royal Garden Party in
+      =< AwardYear.current.year+1
+      | . You must provide the attendee details and your preferred day for attending by
+      =< deadline_for("buckingham_palace_attendees_details", "%l %P, %-d %B %Y.")
+    = link_to "Provide details for Royal Garden Party", edit_palace_invite_url(id: @invite.token), class: "govuk-button govuk-button--start"
+    hr.govuk-section-break.govuk-section-break--l.govuk-section-break--visible
+
+    h2.govuk-heading-m
```
Comment (mauricioac) [inline app/views/group_leader/dashboard/show.html.slim:78]: 'govuk=heading-m' maps to h3 not h2

File: app/views/group_leader/citations/_form.html.slim [hunk 3]
```diff
@@ -0,0 +1,40 @@
+= simple_form_for [:group_leader, @citation] do |f|
+  .govuk-form-group
+    = f.label "Group name", class: "govuk-label govuk-label--s"
```
Comment (mauricioac) [inline app/views/group_leader/citations/_form.html.slim:3]: why do you separate ALL your labels and why do they need this `govuk-label--s` class?

**Review by mauricioac — CHANGES_REQUESTED** [review]

File: app/views/group_leader/citations/_form.html.slim (hunk 3 above)
Comment (TheDancingClown) [reply app/views/group_leader/citations/_form.html.slim:3]: I use govuk-label--s to apply the font weight to match the UI designs. Labels are seperated when I want the different label class or in the case of the citation input, the label should appear above the hints.

**Pull request merged**